src/entity/Living.php

```diff
@@ -659,6 +663,32 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		return $hasUpdate;
 	}
 
+	protected function move(float $dx, float $dy, float $dz) : void{
+		parent::move($dx, $dy, $dz);
+
+		$frostWalkerLevel = $this->armorInventory->getBoots()->getEnchantmentLevel(VanillaEnchantments::FROST_WALKER());
```

Comment:
It might be a good idea to remember this in a property somewhere and update it on armour change. move() is a hot code path, and we don't want to compromise its performance. We already have a listener on the armour inventory for broadcasting updates anyway.

Reply:
Ughh, there is a problem with that, there is no good way to initialize the property. `Living` does not initialize the armor, e.g. if a child class (e.g. Player) spawns with boots, the only way to get the boots would be after child's `initEntity` is executed.

Reply:
I mean, to initialize the property I would have to force the child classes to define it or abuse something like `onFirstUpdate`

Reply:
Ignore the above I had a brain.exe not working moment, is 1am and I have slept very little in the last 24 hours.